docs/lua-api/global-functions/registerhook.md

```diff
@@ -27,7 +29,18 @@ Returns two ids, both of which must be passed to `UnregisterHook` if you want to
 
 ## Example
 ```lua
-RegisterHook("/Script/Engine.PlayerController:ClientRestart", function()
-    print("PlayerController restarted\n")
+PreId, PostId = RegisterHook("/Script/Engine.PlayerController:ClientRestart", function(Context, NewPawn)
+    local playerController = Context:get()
+    local newPawn = NewPawn:get()
+
+    print("PayerController FullName: " .. playerController:GetFullName())
+    if newPawn:IsValid() then
+        print("NewPawn FullName: " .. newPawn:GetFullName())
+    end
```

Comment:
```suggestion
PreId, PostId = RegisterHook("/Script/Engine.PlayerController:ClientRestart", function(Context_Param, NewPawn_Param)
    local PlayerController = Context_Param:get()
    local NewPawn = NewPawn_Param:get()

    print("PayerController FullName: " .. PlayerController:GetFullName())
    if NewPawn:IsValid() then
        print("NewPawn FullName: " .. NewPawn:GetFullName())
    end
```
Potential for some confusion + mixed casing.

I've changed the params to use the `_Param` suffix which serves the purpose of telling the reader that this is a Local/RemoteUnrealParam and thus needs `get()` to be called on it, and it also serves to keep the casing consistent.

An alternative to `_Param`, would be the `In<Name>`/`Out<Name>` style, where ` In` is used for input params and `Out` is used for output params, of which there are none in this example to be clear.

Reply:
It bothers me for a while already but I try to keep the code style similar to what it was before.  
In Lua, it is atypical to use PascalCase for local variables and parameters, it throws me off track every time.   

However, while I agree that local PlayerController and NewPawn should probably be PascalCase, I don't agree with adding special prefixes/suffixes to the names of callbacks variables, simply because it's not coherent with all the other examples and code we use in UEHelpers and other standard mods.

Reply:
We use PascalCase simply because that's what UE uses.
I guess we can keep this PR unchanged and do a larger pass on all the docs later to address this.

Reply:
Unreal Engine is made with modern Microsoft like C++/C# code style.  
In untyped language like lua it's easier to distinguish between global and local variables if they have different casing style.  
While I personally don't like snake case, I use for my own code camelCase for local variables and parameters.